src/wp-includes/widgets/class-wp-widget-media-image.php

```diff
@@ -209,15 +209,15 @@ public function render_media( $instance ) {
 
 			$size = $instance['size'];
 			if ( 'custom' === $size || ! in_array( $size, array_merge( get_intermediate_image_sizes(), array( 'full' ) ), true ) ) {
-				$size = array( $instance['width'], $instance['height'] );
+				$size  = array( $instance['width'], $instance['height'] );
+				$width = $instance['width'];
+			} else {
+				$caption_size = _wp_get_image_size_from_meta( $instance['size'], wp_get_attachment_metadata( $attachment->ID ) );
+				$width        = empty( $caption_size[0] ) ? 0 : $caption_size[0];
 			}
 			$image_attributes['class'] .= sprintf( ' attachment-%1$s size-%1$s', is_array( $size ) ? join( 'x', $size ) : $size );
 
 			$image = wp_get_attachment_image( $attachment->ID, $size, false, $image_attributes );
-
-			$caption_size = _wp_get_image_size_from_meta( $instance['size'], wp_get_attachment_metadata( $attachment->ID ) );
```

Comment:
`$caption_size` is always false when `$instance['size']` is `custom` or any other unregistered image size, so we use the custom `width` to set the `$width` instead.